File: Lab 4/src/shapes/Circle.java
package shapes;

import exceptions.*;

public class Circle extends Shape {
	
	private static int numCircle = 0;
	private double radius;
	private String id;
	
	public Circle( double x, double y, double r, String name ) throws RadiusException
	{
		super( x, y, name );
		id = "C" + ++numCircle;

		if( r < 0 )
		    throw new RadiusException( id );
		else
		    radius = r;
	}
	
	public double get_radius()
	{
		return radius;
	}
	
	public String get_id()
	{
		return id;
	}
	
	public void set_radius( double r ) throws RadiusException
	{
	    if( r < 0 )
		throw new RadiusException( id );
	    else
		radius  = r;
	}
	
	public double area()
	{
		return Math.PI * Math.pow(radius, 2);
	}
	
	public double perimeter()
	{
		return 2 * Math.PI * radius;
	}

    public double volume()
    {
        return 0;
    }
	
	public String toString()
	{
		return "Shape name: " + super.get_name() +
				"\nShape ID: " + this.id +
				"\nOrigin: Point Id: " + super.get_origin().get_id() +
				"\nX_coordinate: " + super.get_origin().get_x() + 
				"\nY_coordinate: " + super.get_origin().get_y() +
				"\nRadius: " + this.radius;
	}
	
}

File: Lab 7/Part III/Circle.cpp
#include <iostream>
using namespace std;

#include <string.h>
#include <math.h>
#include "Point.h"
#include "Shape.h"
#include "Square.h"
#include "Circle.h"
#include "Rectangle.h"
#include "CornerCut.h"
#include "GraphicsWorld.h"

void Circle::f() {}

Circle::Circle(char* name, double x, double y, double r) : Shape(name,x,y)
{
	radius = r;
}

double Circle::area()
{
	return 3.1416 * pow( radius, 2);
}

double Circle::perimeter()
{
	return 2 * 3.1416 * radius;
}

double Circle::getRadius()
{
	return radius;
}

void Circle::setRadius(double r)
{
	radius = r;
}

void Circle::display()
{
	Shape::display();
}
